core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2359,7 +2359,7 @@ private RexNode simplifySearch(RexCall call, RexUnknownAs unknownAs) {
         return RexUtil.expandSearch(rexBuilder, null, call);
       }
     }
-    return call;
+    return call.clone(call.type, ImmutableList.of(a, call.getOperands().get(1)));
```

Comment:
Is there any benefit in not cloning the call if `a` is unchanged?

Reply:
It's certainly more efficient to only clone the call when the search operand is simplified. I have pushed another commit to implement that.